Feature: Convert PHPUnit tests to Pest

mansoorkhan96 opened the pull request:
Hi 👋

This pull request converts PHPUnit tests to Pest. 

Thanks for the amazing work, Team Spatie.

freekmurze commented on tests/ArrayLogger.php:
```diff
@@ -8,47 +8,47 @@ class ArrayLogger implements LoggerInterface
     protected $logLines = [];
 
-    public function emergency($message, array $context = []): void
+    public function emergency(string|\Stringable $message, array $context = []): void
```
Could you import `Stringable`?

mansoorkhan96 replied:
Done 👍

freekmurze commented on tests/ImageTest.php:
```diff
@@ -1,43 +1,29 @@
-
-        new Image('non existing file');
-    }
+it('will throw an exception when given a non existing file', fn () => new Image('non existing file'))
```
Let's use a normal function instead of a short closure here.

freekmurze reviewed (changes requested):
Very nice! I only got a few nitpicks.

mansoorkhan96 commented:
Hey, @freekmurze I have updated the PR.

freekmurze commented:
It seems like every tests is failing. Please check locally and push a fix.

mansoorkhan96 commented:
Changes description:

- Removed type hinting for `$message` argument in `ArrayLogger.php` for backward compatibility tests.
- Updated expectations in `OptimizerTest` to be in Unix style. (Apologies, I am on windows and did not notice this issue).
- Replaced short closure functions with normal functions for backward compatibility tests.

mansoorkhan96 commented:
Hey, Sir @freekmurze 

Does this require further changes? Please let me know when you have time :)

Thanks.

freekmurze commented:
Looks good, thanks!
